pydarn/plotting/fan.py

```diff
@@ -0,0 +1,243 @@
+                            dmap_data[plot_beams[0][0]]['time.hr'], dmap_data[plot_beams[0][0]]['time.mt'],
+                            dmap_data[plot_beams[0][0]]['time.sc'])
+        for x in range(fan_shape[0]):
+            for y in range(fan_shape[1]):
+                # Convert to AACGM
+                geomag = np.array(aacgmv2.
+                                  get_aacgm_coord(beam_corners_lats[x, y],
+                                                  beam_corners_lons[x, y],
+                                                  250, dtime))
+
+                beam_corners_aacgm_lats[x, y] = geomag[0]
+                beam_corners_aacgm_lons[x, y] = geomag[1]
+
+        # Work out shift due in MLT
+        mltshift = beam_corners_lons[x, y] - \
+            (aacgmv2.convert_mlt(beam_corners_lons[x, y], dtime) * 15)
+        beam_corners_mlts = beam_corners_aacgm_lons - mltshift
+
+        # Hold the beam positions
+        thetas = np.radians(beam_corners_mlts)
+        rs = beam_corners_aacgm_lats
+
+        # Get range-gate data and groundscatter array for given scan
+        scan = np.zeros((fan_shape[0] - 1, fan_shape[1]-1))
+        grndsct = np.zeros((fan_shape[0] - 1, fan_shape[1]-1))
+        iterat = 0
+        for i in np.nditer(plot_beams):
+            try:
+                slist = dmap_data[i.astype(int)]['slist']
+                scan[slist, iterat] = dmap_data[i.astype(int)][parameter]
+                grndsct[slist, iterat] = dmap_data[i.astype(int)]['gflg']
+                iterat += 1
+            # if there is no slist field this means partial record
+            except KeyError:
+                continue
+
+        # Colour table and max value selection depending on parameter plotted
+        # Load defaults if none given
+        # TODO: use cmaps as over writting cmap is bad practice...
+        # did I do that in my code ... hmm
+        if cmap is None:
+            cmap = {'p_l': 'plasma', 'v': PyDARNColormaps.PYDARN_VELOCITY,
+                    'w_l': PyDARNColormaps.PYDARN_VIRIDIS,
+                    'elv': PyDARNColormaps.PYDARN}
+            cmap = plt.cm.get_cmap(cmap[parameter])
+
+        # Setting zmin and zmax
+        defaultzminmax = {'p_l': [0, 50], 'v': [-200, 200],
+                          'w_l': [0, 250], 'elv': [0, 50]}
+        if zmin is None:
+            zmin = defaultzminmax[parameter][0]
+        if zmax is None:
+            zmax = defaultzminmax[parameter][1]
+
+        # Setup plot
+        # TODO: eek! remeber there is an option to pass in a variable called ax
+        # This may screw up references
+        ax = plt.axes(polar=True)
+        if beam_corners_aacgm_lats[0,0] > 0:
+            ax.set_ylim(90, lowlat)
+            ax.set_yticks(np.arange(lowlat, 90, 10))
+        else:
+            ax.set_ylim(-90, -lowlat)
+            ax.set_yticks(np.arange(-lowlat, -90, -10))
+        ax.set_xticklabels(['00', '', '06', '', '12', '', '18', ''])
```

Comment:
Recommend using the style '18:00' or '1800'.  What do others think?

Reply:
I've always been used to it the way it's written, but open to change if people would rather the hourly format.